### ISSUE TITLE
Unable to add ability to the actor.

### ISSUE THREAD

**Issue by jdmesalosada** [jan-molak/serenity-js#119]
Hi, I'm trying to set the ability to interact with a database to an actor, but I'm getting the following error:

**"I don't have the ability to InteractWithDatabase, said Brayan sadly."**

This is my Ability:

```
import { Ability, UsesAbilities } from "@serenity-js/core/lib/screenplay";
const sql = require('mssql')

const config = {
  ///config info here
};

export class InteractWithDatabase implements Ability {

  static using() {
    return new InteractWithDatabase();
  }

  static as(actor: UsesAbilities): InteractWithDatabase {
    debugger
    return actor.abilityTo(InteractWithDatabase);
  }

  constructor() {
  }

  execute(query: string): PromiseLike<any> {
    return sql.connect(config).then(() => {
      return sql.query`${query}`
    }).then(result => {
      console.dir(result)
    }).catch(err => {
      // ... error checks
    })
  }
}

```

This is the interaction:

```
import { Interaction, UsesAbilities } from "@serenity-js/core/lib/screenplay";
import { InteractWithDatabase } from "./InteractWithDatabase";

export class ExecuteQuery implements Interaction {

  static using(query: string) {
    return new ExecuteQuery(query);

  }

  performAs(actor: UsesAbilities): PromiseLike<any> {
    return InteractWithDatabase.as(actor).execute(this.query);
  }

  constructor(private query: string) {
  }
}

```

This is my task:

```
export class GetUserData implements Task {

  static fromDatabase() {
    return new GetUserData();
  }

  performAs(actor: PerformsTasks): PromiseLike<void> {
    return actor.attemptsTo(
      ExecuteQuery.using('query here'),
    );
  }
}
```
I tried adding the ability in the actors class:

```
export class Actors implements Cast {

    abilities = [BrowseTheWeb.using(protractor.browser), InteractWithDatabase.using()];

    actor(name: string): Actor {
        debugger
        return Actor.named(name).whoCan(this.abilities)
    }
}

```

but doesn't work. :(

I hope someone can help me with this.

Thanks

**Comment by jan-molak** [comment]
Hey @jdmesalosada,

When you instantiate the actor you give it a list of abilities:

```typescript
const actor = Actor.named('Brayan').whoCan([BrowseTheWeb.using(protractor.browser), InteractWithDatabase.using()]);
```

while the `whoCan(...)` method is variadic, so you can either say:

```typescript
const actor = Actor.named('Brayan').whoCan(BrowseTheWeb.using(protractor.browser), InteractWithDatabase.using());
```

or de-structure the list:

```typescript
export class Actors implements Cast {

    abilities = [BrowseTheWeb.using(protractor.browser), InteractWithDatabase.using()];

    actor(name: string): Actor {
        return Actor.named(name).whoCan(...this.abilities)
    }
}
```

Other than that it looks good to me!

You might want to consider using the shorthand notation for interactions:

```typescript
const ExecuteQuery = (query: string) => Interaction.where(`#actor executes query '${query}`, 
    actor => InteractWithDatabase.as(actor).execute(query)
);
```

Also, what are you planning to do with the data returned by the query? If you want to assert on it, you might want to create a question:

```typescript
const UserData = () => Question.about(`user data stored in the database`, 
    actor => InteractWithDatabase.as(actor).execute(query)
);
```

which you can then execute using `See.if`:

```typescript
actor.attemptsTo(
    See.if(UserData(), data => expect(data).to.eventually.deep.equal(...))
);
```

Let me know if that helps!

Jan

**Comment by jdmesalosada** [comment]
Hi @jan-molak excellent explanation. My idea with the data I got from database is to compare it with something in the UI. For example I got the info for the user with id 123, so I create a query to get the info for that user. In the info I got from database there is a field called username. When the user is logged into the application the username should be displayed in the UI, so I have to create a question in order to capture the username in the UI.

I have to compare that the value of the field username in the database matches with the value displayed in the UI.

With the ExecuteQuery like a question I should compare two questions right?
 I'm not sure how can achieve this.

Thanks.
